### DISCUSSION TITLE
Improving Publish Test Results action

### DISCUSSION THREAD

**Discussion by gntpet** [microsoft/AL-Go#1598] [Ideas]
Hey @freddydk , 

I been recently playing with improving **publish-unit-test-result-action** from this [repo](https://github.com/EnricoMi/publish-unit-test-result-action). There are more actions, doing the similar stuff. I chose to picked repo from EnricoMi, since it's allowed within our enterprise organization.

It has some advantages over AL-GO way of doing, since it's publishing it in more places:

![image](https://github.com/microsoft/AL-Go/assets/16342128/ad3f5119-346f-4543-9f05-116753689b53)

I hade to make these two small changes in a pipeline: add required permissions & add new action

![image](https://github.com/microsoft/AL-Go/assets/16342128/8fea7a10-04f2-4a83-928c-4c29e8fcfcaf)

If I would like to implement this one, instead of standard Al-Go ones, what are my options? I don't want to loose system updates.

I also wonder if you considered using off the shelf actions for publish test results. What was the reason, you choose to build a custom one?

Best Regards, 
Gintautas

**Comment by freddydk** [comment]
I will take this as requirement to the indirect template / customization functionality.
Microsoft cannot include 3rd party actions in AL-Go, but it would be nice to be able for partners to use other apps / actions in their repositories.

**Comment by gntpet** [comment]
There's one more idea that we would like to implement. It is to add notification steps (to Teams or Slack) for deployments and  testresults. Can you take this requirement into account as well? 

Is this template customization functionality planned already? any rough ETA?

**Comment by freddydk** [comment]
It is work in progress, but something that has been standing still for a while

https://github.com/microsoft/AL-Go/pull/853

Hoping to have time to look more into this over the summer

**Comment by freddydk** [comment]
It is the next thing I will throw my attention to.

**Comment by KristofKlein** [comment]
I was looking into [Allure Report](https://allurereport.org/) the other day. That can collect the testresult and render them into a report, that it than can upload as a gitHub page. incl. history. I understrand MS does not want 3rd Party actions. Still this might be a path for the one or another... :)
